src/lib.rs

```diff
@@ -69,7 +70,7 @@ impl HypersyncClient {
             .context("run inner collect")
             .map_err(map_err)?;
 
-        convert_response(resp)
+        convert_response(resp, ShouldChecksum(self.enable_checksum_addresses))
```

Comment:
should just use bool

Reply:
I'm happy to change it.

But I'm using the "new type" idiom. Which IMO is the best way to express flag arguments in rust with no overhead. https://rust-unofficial.github.io/patterns/patterns/behavioural/newtype.html

Reply:
It is something good to do but in the end it adds cognitive load when reading code with no practical benefit, for me at least